### DISCUSSION TITLE
hetzner-k3s is BY FAR the fastest tool in the market at creating a Kubernetes cluster! 50 nodes HA cluster created in just 2 minutes and 42 seconds 😂

### DISCUSSION THREAD

**Discussion by vitobotta** [vitobotta/hetzner-k3s#347] [Announcements]
I still can't believe my eyes what I just saw 😂 I am not kidding guys. I did a huge refactoring and improved concurrency significantly, so I was able to create a 50 node cluster (3 master for HA control plane plus 47 workers) in well under 3 minutes!

I believe that there is no other tool in the market that can create a cluster of this size this quickly :D

These changes will come together with various improvements with next release.

**Comment by vitobotta** [comment]
With further improvements, yesterday I was able to create a 200 node cluster in just under 4 minutes :D

**Reply by adrianalter** [reply]
you are amazing

**Reply by vitobotta** [reply]
Thanks, I should have perhaps updated this thread because the new record, from yesterday is **300 nodes** in **4 minutes, 59 seconds** :D
